### PR TITLE
[Dask] Deprecate `with_requests`

### PR DESCRIPTION
**Pull request by AlonMaor14**
Deprecate dask `with_requests` since it's long overdue
Use `with_scheduler_requests` and `with_worker_requests` instead

### CODE REVIEW

File: mlrun/runtimes/daskjob.py [hunk 1]
```diff
@@ -602,7 +596,7 @@ def enrich_dask_cluster(function, secrets, client_version):
         env.append(spec.extra_pip)
 
     pod_labels = get_resource_labels(function, scrape_metrics=config.scrape_metrics)
-    worker_args = ["dask-worker", "--nthreads", str(spec.nthreads)]
+    worker_args = ["dask worker", "--nthreads", str(spec.nthreads)]
```
Comment (Tankilevitch) [inline mlrun/runtimes/daskjob.py:599]: I wonder what is the effect of this change... could it be that it never worked properly?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
General comment regarding the naming you chose, it is unclear what is protocol, I would use something like `pickle_protocol` or `mlrun_pickle` which describes that we are referring to some implementation of pickle.

File: mlrun/runtimes/daskjob.py (hunk 1 above)
Comment (AlonMaor14) [reply mlrun/runtimes/daskjob.py:599]: this is about to be deprecated. I will add a todo and make the change in another PR

File: mlrun/utils/pickle.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+else:
+    import pickle
+
+
+def dumps(x):
+    return pickle.dumps(x)
+
+
+def loads(x):
+    return pickle.loads(x)
```
Comment (guy1992l) [inline mlrun/utils/pickle.py:33]: not sure why we would need to use `dumps` and `loads` from here instead of directly use the pickle imported here :)

File: mlrun/utils/pickle.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+# import pickle5 on clients with python < 3.8 since the server is using pickle protocol 5 (python 3.9)
+# TODO: remove this when we drop support for python 3.7
+if sys.version_info < (3, 8):
+    try:
+        import pickle5 as pickle
+    except ImportError:
+        import pickle
+else:
+    import pickle
```
Comment (guy1992l) [inline mlrun/utils/pickle.py:25]: I think `cloudpickle` can handle this stuff automatically, and since it is in our requirements we could be simply using it instead... you should try it out :)

File: mlrun/runtimes/daskjob.py [hunk 4]
```diff
@@ -465,17 +465,11 @@ def with_worker_limits(
...
+            "Dask's with_requests is deprecated and will be removed in 1.4.0, use "
             "with_scheduler_requests/with_worker_requests instead",
-            # TODO: In 0.8.0 deprecate and replace with_requests to with_worker/scheduler_requests in examples & demos
-            #  (or maybe just change behavior ?)
-            PendingDeprecationWarning,
         )
-        # the scheduler/worker specific function was introduced after the general one, to keep backwards compatibility
-        # this function just sets the requests for both of them
-        self.with_scheduler_requests(mem, cpu)
-        self.with_worker_requests(mem, cpu)
```
Comment (guy1992l) [inline mlrun/runtimes/daskjob.py:478]: how are these related?

**Review by guy1992l — COMMENTED** [review]
I should change in `mlrun.frameworks` to `cloudpickle` and I think in general `mlrun` should only use it. I'm not a pickling master, so you should try it out and see if it resolved the pickle to pickle 5 issues on its own.

File: mlrun/utils/pickle.py (hunk 2 above)
Comment (AlonMaor14) [reply mlrun/utils/pickle.py]: removed
